src/java/htsjdk/samtools/SAMBinaryTagAndValue.java

```diff
@@ -73,6 +74,34 @@ private boolean typeSafeEquals(final SAMBinaryTagAndValue that) {
         }
     }
 
+    private boolean valueEquals(SAMBinaryTagAndValue that) {
+        if (this.value == null) {
+            return that.value == null;
+        }
+        else if (that.value == null) {
+            return false;
+        }
+        else if (this.value instanceof byte[]) {
+            return that.value instanceof byte[] ?
+                Arrays.equals((byte[])this.value, (byte[])that.value) : false;
+        }
+        else if (this.value instanceof short[]) {
+            return that.value instanceof short[] ?
+                    Arrays.equals((short[])this.value, (short[])that.value) : false;
+        }
+        else if (this.value instanceof int[]) {
+            return that.value instanceof int[] ?
+                    Arrays.equals((int[])this.value, (int[])that.value) : false;
+        }
+        else if (this.value instanceof float[]) {
+            return that.value instanceof float[] ?
+                    Arrays.equals((float[])this.value, (float[])that.value) : false;
+        }
+        else {
+            return this.value.equals(that.value);
```

Comment:
should this include a `this.value instanceOf Object[]` case?   What happens with an `Integer[]`?.  Based on the comment in `cloneValue()` it seems like Integer[] is possible, but `Array.equals` is an identity comparison I believe, not an element by element equality.